services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -535,46 +544,67 @@ async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
-            stop=stop_after_attempt(3),
-            wait=wait_exponential(),
+            # waits exponentially to a max of `stop_delay` seconds
+            stop=stop_after_delay(stop_delay),
+            wait=wait_exponential(min=1),
             retry=retry_if_exception_type(_RetryError),
             reraise=True,
         ):
             with attempt:
                 try:
                     # fetch information from service name
-                    service_inspect = await client.services.inspect(
-                        scheduler_data.service_name
-                    )
+                    service_inspect = await client.services.inspect(service_name)
                     service_version = service_inspect["Version"]["Index"]
                     service_id = service_inspect["ID"]
                     spec = service_inspect["Spec"]
 
-                    spec["Labels"][
-                        DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL
-                    ] = scheduler_data.as_label_data()
+                    updated_spec = nested_update(
+                        spec,
+                        update_in_service_spec,
+                        include=get_longest_key_paths(update_in_service_spec),
+                    )
 
                     await client._query_json(  # pylint: disable=protected-access
                         f"services/{service_id}/update",
                         method="POST",
-                        data=json.dumps(clean_map(spec)),
+                        data=json.dumps(clean_map(updated_spec)),
                         params={"version": service_version},
                     )
                 except aiodocker.exceptions.DockerError as e:
-                    if not (
+                    if (
                         e.status == status.HTTP_404_NOT_FOUND
-                        and e.message
-                        == f"service {scheduler_data.service_name} not found"
+                        and e.message == f"service {service_name} not found"
                     ):
-                        raise e
-                    if (
+                        log.debug(
+                            "Skip update for service '%s' which could not be found.",
+                            service_name,
+                        )
+                    elif (
                         e.status == status.HTTP_500_INTERNAL_SERVER_ERROR
                         and e.message
                         == "rpc error: code = Unknown desc = update out of sequence"
                     ):
                         raise _RetryError() from e
-                    log.debug(
-                        "Skip update for service '%s' which could not be found",
-                        scheduler_data.service_name,
-                    )
+                    else:
+                        raise e
+
+
+async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
+    await _update_service_spec(
```

Comment:
Q? why do you create a client for every call? Wouldn't it make more sense to create a single client for the app lifespam ?